Gulp: Make sure we tear down servers and simulators

JohanLorenzo opened the pull request:
Fixes #122

JohanLorenzo commented:
I found a way to run some tasks even though the previous failed. In order to show it works, here are:

* [A patch](https://github.com/fxbox/app/pull/123/commits/c68cd976b5826cfaed3e01ae6ebc74c574aa1512) that makes the [integration tests hanging](https://travis-ci.org/fxbox/app/builds/125512278#L425).
* The fix on top the error => the error is [immediately reported](https://travis-ci.org/fxbox/app/builds/125512980#L413). 
* The fix alone, with the [tests passing](https://travis-ci.org/fxbox/app/builds/125513496). 

r? @gmarty

JohanLorenzo commented on gulpfile.js:
```diff
@@ -241,9 +241,9 @@ gulp.task('run-test-integration', function() {
-  return runSequence(
-    'start-simulators', 'run-test-integration', 'stop-simulators', cb
-  );
+  runSequence('start-simulators', 'run-test-integration', function tearDown() {
```
I know, the function can remain anonymous, but I find the intent of wrapping the callback clearer with that kind of names

gmarty replied:
nit: can you use arrow function whenever possible?

gmarty commented on gulpfile.js:
```diff
@@ -257,8 +257,9 @@ gulp.task('test', function(cb) {
-    cb
+    function tearDown() {
+      runSequence('stop-webserver', cb);
+    }
```
nit: same here.

gmarty commented:
I prefer arrow functions, specially in a Gulp file, over named functions. Maybe add a comment to explain why you're doing it.
But r+.

JohanLorenzo commented:
I avoided to use arrow function for a matter of consistency. None is used within the entire gulpfile. I don't mind using them, though. I updated my PR and added comment in the code.
